### PR TITLE
Implement azure maps weather

### PR DESCRIPTION
**Pull request by zhz0704**
Implementation of Azure.Maps.Weather

### CODE REVIEW

File: sdk/maps/Azure.Maps.Weather/README.md
Comment (dubiety) [inline sdk/maps/Azure.Maps.Weather/README.md:132]: All examples should avoid using `var`, please replace them with actual type names.

File: sdk/maps/Azure.Maps.Weather/src/MapsWeatherClient.cs
Comment (dubiety) [inline sdk/maps/Azure.Maps.Weather/src/MapsWeatherClient.cs:4]: Remove this line

File: sdk/maps/Azure.Maps.Weather/src/MapsWeatherClient.cs
Comment (dubiety) [inline sdk/maps/Azure.Maps.Weather/src/MapsWeatherClient.cs:44]:
```suggestion
        /// <param name="credential"> Shared key credential used to authenticate to an Azure Maps Weather Service. </param>
```

Comment (dubiety) [reply sdk/maps/Azure.Maps.Weather/src/MapsWeatherClient.cs:44]: Please also update the typo in other docstrings `Render` -> `Weather`

File: sdk/maps/Azure.Maps.Weather/src/MapsWeatherClient.cs
Comment (dubiety) [inline sdk/maps/Azure.Maps.Weather/src/MapsWeatherClient.cs:137]: Is this constructor used anywhere? If not, we can remove it.

File: sdk/maps/Azure.Maps.Weather/src/MapsWeatherClient.cs
Comment (dubiety) [inline sdk/maps/Azure.Maps.Weather/src/MapsWeatherClient.cs:146]: What does `[Protocol Method]` mean here?

File: sdk/maps/Azure.Maps.Weather/src/MapsWeatherClient.cs
Comment (dubiety) [inline sdk/maps/Azure.Maps.Weather/src/MapsWeatherClient.cs:159]: Can we rename it to `GetHourlyWeatherForecastAsync` and `GetHourlyWeatherForecast` ? It's not very clear to me at the first glance of what hourly forecast these methods want to forecast.

File: sdk/maps/Azure.Maps.Weather/src/MapsWeatherClient.cs
Comment (dubiety) [inline sdk/maps/Azure.Maps.Weather/src/MapsWeatherClient.cs:150]: What's this? Wrong copy?

File: sdk/maps/Azure.Maps.Weather/src/MapsWeatherClient.cs
Comment (dubiety) [reply sdk/maps/Azure.Maps.Weather/src/MapsWeatherClient.cs:159]: Same for `GetMinuteForecastAsync`, `GetMinuteForecast`, `GetQuarterDayForecastAsync`, `GetQuarterDayForecast`

File: sdk/maps/Azure.Maps.Weather/src/MapsWeatherClient.cs
Comment (dubiety) [reply sdk/maps/Azure.Maps.Weather/src/MapsWeatherClient.cs:150]: Same for other methods

File: sdk/maps/Azure.Maps.Weather/src/MapsWeatherClient.cs
Comment (dubiety) [inline sdk/maps/Azure.Maps.Weather/src/MapsWeatherClient.cs:375]: Consider rename it to `GetCurrentWeatherConditions(Async)`

File: sdk/maps/Azure.Maps.Weather/src/MapsWeatherClient.cs
Comment (dubiety) [inline sdk/maps/Azure.Maps.Weather/src/MapsWeatherClient.cs:388]: Can we assign `options.Details` a default value to `true` in `GetCurrentConditionsOptions.cs`?
Something like: `public bool? Details { get; set; } = true;`?
And here the statement can be simplified as:
`options.Details?.Value ? "true" : "false"`

File: sdk/maps/Azure.Maps.Weather/src/MapsWeatherClientOptions.cs
Comment (dubiety) [inline sdk/maps/Azure.Maps.Weather/src/MapsWeatherClientOptions.cs:41]: Use `<see href="">xxx</see>` for hyperlink

File: sdk/maps/Azure.Maps.Weather/src/Models/Options/GetAirQualityHourlyForecastsOptions.cs
Comment (dubiety) [inline sdk/maps/Azure.Maps.Weather/src/Models/Options/GetAirQualityHourlyForecastsOptions.cs:14]: For measurable class member (time/distance/length/temperature), we should always clarify the unit. I suggest to rename it to `DurationInHours`

Comment (dubiety) [reply sdk/maps/Azure.Maps.Weather/src/Models/Options/GetAirQualityHourlyForecastsOptions.cs:14]: Same for other `Duration` member variable in other `Options` classes

File: sdk/maps/Azure.Maps.Weather/src/Models/Options/GetCurrentConditionsOptions.cs
Comment (dubiety) [inline sdk/maps/Azure.Maps.Weather/src/Models/Options/GetCurrentConditionsOptions.cs:16]: Suggest to rename it to `ShouldIncludeDetails`.

File: sdk/maps/Azure.Maps.Weather/src/Models/Options/GetDailyIndicesOptions.cs
Comment (dubiety) [inline sdk/maps/Azure.Maps.Weather/src/Models/Options/GetDailyIndicesOptions.cs:15]: Use `<see href="">xxx</see>`

File: sdk/maps/Azure.Maps.Weather/src/Models/Options/GetMinuteForecastOptions.cs
Comment (dubiety) [inline sdk/maps/Azure.Maps.Weather/src/Models/Options/GetMinuteForecastOptions.cs:14]: Rename it to `IntervalInMinutes`

File: sdk/maps/Azure.Maps.Weather/src/Models/Options/GetTropicalStormForecastOptions.cs
Comment (dubiety) [inline sdk/maps/Azure.Maps.Weather/src/Models/Options/GetTropicalStormForecastOptions.cs:16]: We should create a `BasinIdEnum` for users.

File: sdk/maps/Azure.Maps.Weather/src/Models/Options/GetDailyIndicesOptions.cs
Comment (dubiety) [inline sdk/maps/Azure.Maps.Weather/src/Models/Options/GetDailyIndicesOptions.cs:15]:
```suggestion
        /// <summary> Numeric index identifier that can be used for restricting returned results to the corresponding index type. Cannot be paired with <c>indexGroupId</c>. Please refer to [Weather services in Azure Maps](/azure/azure-maps/weather-services-concepts#index-ids-and-index-groups-ids) for details and to see the supported indices. </summary>
```

File: sdk/maps/Azure.Maps.Weather/src/Models/WeatherAlongRouteWaypoint.cs
Comment (dubiety) [inline sdk/maps/Azure.Maps.Weather/src/Models/WeatherAlongRouteWaypoint.cs:14]: `ETA` -> `EtaInMinutes`

File: sdk/maps/Azure.Maps.Weather/tests/MapsWeatherTests.cs
Comment (dubiety) [inline sdk/maps/Azure.Maps.Weather/tests/MapsWeatherTests.cs:12]: Let's put similar namespaces together. Please help to move `System.Runtime.InteropServices` in line 7.

File: sdk/maps/Azure.Maps.Weather/tests/Samples/WeatherClientSamples.cs
Comment (dubiety) [inline sdk/maps/Azure.Maps.Weather/tests/Samples/WeatherClientSamples.cs:15]: Please put similar namespace together

**Comment by dubiety** [comment]
We should also hide all `GeoJsonxxx` classes (Make them internal classes), for example: GeoJsonPoint, GeoJsonMultiPoint, GeoJsonLineString, GeoJsonMultiLineString, GeoJsonPolygon, GeoJsonMultiPolygon, GeoJsonGeometryCollection, GeoJsonFeature, GeoJsonFeatureCollection

File: sdk/maps/Azure.Maps.Weather/src/Generated/Models/GeoJsonObject.cs
Comment (dubiety) [inline sdk/maps/Azure.Maps.Weather/src/Generated/Models/GeoJsonObject.cs:15]: This should be internal

File: sdk/maps/Azure.Maps.Weather/src/Generated/Models/GeoJsonObjectType.cs
Comment (dubiety) [inline sdk/maps/Azure.Maps.Weather/src/Generated/Models/GeoJsonObjectType.cs:11]: This should be internal

File: sdk/maps/Azure.Maps.Weather/src/Generated/Models/StormWindRadiiSummary.cs
Comment (dubiety) [inline sdk/maps/Azure.Maps.Weather/src/Generated/Models/StormWindRadiiSummary.cs:48]: We need to replace `GeoJsonGeometry` with `Azure.Core.GeoJson` `GeoObject`

Comment (dubiety) [reply sdk/maps/Azure.Maps.Weather/src/Generated/Models/StormWindRadiiSummary.cs:48]: Please also try to make a test or sample to ensure this works after replacing.

File: sdk/maps/Azure.Maps.Weather/tests/MapsWeatherTests.cs
Comment (dubiety) [inline sdk/maps/Azure.Maps.Weather/tests/MapsWeatherTests.cs:259]: Suggest to use [string interpolation](https://learn.microsoft.com/en-us/dotnet/csharp/tutorials/exploration/interpolated-strings-local) expressions `$"print {varName}"` or [composite formatting](https://learn.microsoft.com/en-us/dotnet/standard/base-types/composite-formatting): `Console.Writeline("{0}, {1}", lat, long)`

File: sdk/maps/Azure.Maps.Weather/src/Models/WeatherWindow.cs
Comment (dubiety) [inline sdk/maps/Azure.Maps.Weather/src/Models/WeatherWindow.cs:4]: Remove this line

File: sdk/maps/ci.yml
Comment (dubiety) [inline sdk/maps/ci.yml:48]: - We should remove these 2 lines before merging. Otherwise, this can block other packages to ship.

File: sdk/maps/Azure.Maps.Weather/src/Models/GeoJsonFeature.cs
Comment (dubiety) [inline sdk/maps/Azure.Maps.Weather/src/Models/GeoJsonFeature.cs:11]: We don't need `CodeGenModel` if we just want to make it internal. `CodeGenModel` is for renaming the class.

File: sdk/maps/Azure.Maps.Weather/src/Models/StormWindRadiiSummary.cs
Comment (dubiety) [inline sdk/maps/Azure.Maps.Weather/src/Models/StormWindRadiiSummary.cs:14]: We don't need `CodeGenModel` if we just want to change its members? `CodeGenModel` is for renaming the class.

**Comment by dubiety** [comment]
Overall it looks good! I'll approve it when all the comments are resolved. Thanks for the great effrot, @zhz0704 !

**Comment by zhz0704** [comment]
Done :)

**Review by dubiety — APPROVED** [review]

**Pull request merged**
